tests/TestCase/Database/Type/FloatTypeTest.php

```diff
@@ -194,4 +198,60 @@ public function testToStatement(): void
     {
         $this->assertSame(PDO::PARAM_STR, $this->type->toStatement('', $this->driver));
     }
+
+    public function testMarshalDefaultsToNull(): void
+    {
+        $table = $this->getTableLocator()->get('Datatypes');
+        $validator = new Validator()->numeric('floaty');
+        $table->setValidator('default', $validator);
+
+        // Testing a invalid patchEntity with a validator making floaty required
+        $newEntity = $table->newEmptyEntity();
+        $newEntity = $table->patchEntity($newEntity, [
+            'floaty' => null,
+        ]);
+        $this->assertFalse($table->save($newEntity));
+        $this->assertEquals([
+            'floaty' => [
+                '_empty' => 'This field cannot be left empty',
+            ],
+        ], $newEntity->getErrors());
+
+        // Saving via a direct patch should work as there is no validation or marshaling
+        $newEntity = $table->newEmptyEntity();
+        $newEntity = $newEntity->patch([
+            'floaty' => null,
+        ]);
+        $this->assertInstanceOf(EntityInterface::class, $table->save($newEntity));
+        $this->assertNull($newEntity->floaty);
+
+        // Saving via a direct patch with an invalid floaty also works as the string is cast to float
+        $newEntity = $table->newEmptyEntity();
+        $newEntity = $newEntity->patch([
+            'floaty' => 'invalid',
+        ]);
+        $newEntity = $table->save($newEntity);
+        $this->assertInstanceOf(EntityInterface::class, $newEntity);
+        $this->assertEquals('invalid', $newEntity->floaty); // which is weird but ok
```

Comment:
It would be better if the entity reflected the state of the database after the save is complete. That would require us to reload the entity though.

Reply:
> That would require us to reload the entity though.

Wouldn't it be enough if we set the value from the `toDatabase()` method of the used DB Type into the entity field back?

But I guess this only works for certain DB types as e.g. the DateTime type passes a formated string to the DB but we need a Date object inside the entity.